src/libraries/System.Private.CoreLib/src/System/Random.cs

```diff
@@ -218,7 +218,73 @@ public virtual int Next(int maxValue)
             return (int)(Sample() * maxValue);
         }
 
-        /*=====================================Next=====================================
+        private long FullLong()
+        {
+            long result = 0;
+
+            for (int i = 0; i < 4; i++)
+            {
+                short part;
+                unchecked
+                {
+                    part = (short) InternalSample();
+                }
+
+                result |= (long) part;
+                result <<= 16;
+            }
+
+            return result;
+        }
+
+        /*==================================NextInt64===================================
+        **Returns: An long [0..long.maxValue)
+        **Arguments: maxValue -- One more than the greatest legal return value.
+        **Exceptions: None.
+        ==============================================================================*/
+        public virtual long NextInt64()
+        {
+            return FullLong();
+        }
+
+        /*==================================NextInt64===================================
+        **Returns: A long [0..maxValue)
+        **Arguments: maxValue -- One more than the greatest legal return value.
+        **Exceptions: None.
+        ==============================================================================*/
+        public virtual long NextInt64(long maxValue)
+        {
+            if (maxValue < 0)
+                throw new ArgumentOutOfRangeException(nameof(maxValue), SR.Format(SR.ArgumentOutOfRange_MustBePositive, nameof(maxValue)));
+
+            if (maxValue == 0)
+                return 0;
+
+            long fullLong = (long) FullLong();
+            return (long.MaxValue & fullLong) % maxValue;
```

Comment:
This distorts results, for two reasons:

 1. The bitwise-and, possibly in an attempt to make the range positive, subtly distorts the number of results, due to there being one more negative number than there are positive (or two, if `long.MaxValue` is never returned).  If  `FullLong()` returned the correct range, the bitwise-and should be unnecessary.
 1. The modulus operator distorts the results, because usually `maxValue` will not be an integral factor of the range of the random.  [See this for a discussion, and the potential fix if modulus will continue to be used](https://stackoverflow.com/questions/10984974/why-do-people-say-there-is-modulo-bias-when-using-a-random-number-generator).